### PR TITLE
Changed API call URLs to v2 version

### LINKED ISSUE #2328: Use API v2 in frontend calls.
When `two-databases` branch is merged into `master`, let's try to use v2 endpoints. 

This a list of endpoints of `v2`. `FE` may not use all of them. 
Basically, `v2` adds the prefix `/api/v2` to `v1` urls:
 
This a list of endpoints of `v2`: 
- /api/v2/asset_snapshots/
- /api/v2/asset_snapshots/`<uid>`/
- /api/v2/asset_snapshots/`<uid>`/preview/
- /api/v2/asset_snapshots/`<uid>`/xform/
- /api/v2/assets/
- /api/v2/assets/`<parent_lookup_asset>`/data/
- /api/v2/assets/`<parent_lookup_asset>`/data/`<pk>`/
- /api/v2/assets/`<parent_lookup_asset>`/data/`<pk>`/edit/
- /api/v2/assets/`<parent_lookup_asset>`/data/`<pk>`/validation_status/
- /api/v2/assets/`<parent_lookup_asset>`/data/bulk/
- /api/v2/assets/`<parent_lookup_asset>`/data/validation_statuses/	
- /api/v2/assets/`<parent_lookup_asset>`/files/
- /api/v2/assets/`<parent_lookup_asset>`/files/`<uid>`/
- /api/v2/assets/`<parent_lookup_asset>`/files/`<uid>`/content/	
- /api/v2/assets/`<parent_lookup_asset>`/hook-signal/	
- /api/v2/assets/`<parent_lookup_asset>`/hooks/
- /api/v2/assets/`<parent_lookup_asset>`/hooks/`<parent_lookup_hook>`/logs/	
- /api/v2/assets/`<parent_lookup_asset>`/hooks/`<parent_lookup_hook>`/logs/`<uid>`/
- /api/v2/assets/`<parent_lookup_asset>`/hooks/`<parent_lookup_hook>`/logs/`<uid>`/retry/	
- /api/v2/assets/`<parent_lookup_asset>`/hooks/`<uid>`/	
- /api/v2/assets/`<parent_lookup_asset>`/hooks/`<uid>`/retry/	
- /api/v2/assets/`<parent_lookup_asset>`/permissions/	
- /api/v2/assets/`<parent_lookup_asset>`/permissions/`<uid>`/
- /api/v2/assets/`<parent_lookup_asset>`/permissions/bulk/	
- /api/v2/assets/`<parent_lookup_asset>`/permissions/clone/
- /api/v2/assets/`<parent_lookup_asset>`/versions/
- /api/v2/assets/`<parent_lookup_asset>`/versions/`<uid>`/
- /api/v2/assets/`<uid>`/
- /api/v2/assets/`<uid>`/content/
- /api/v2/assets/`<uid>`/deployment/
- /api/v2/assets/`<uid>`/koboform/
- /api/v2/assets/`<uid>`/table_view/
- /api/v2/assets/`<uid>`/valid_content/
- /api/v2/assets/`<uid>`/xform/
- /api/v2/assets/`<uid>`/xls/
- /api/v2/assets/hash/
- /api/v2/collections/
- /api/v2/collections/`<parent_lookup_collection>`/permissions/
- /api/v2/collections/`<parent_lookup_collection>`/permissions/`<uid>`/
- /api/v2/collections/`<parent_lookup_collection>`/permissions/bulk/
- /api/v2/collections/`<parent_lookup_collection>`/permissions/clone/
- /api/v2/collections/`<uid>`/
- /api/v2/permissions/
- /api/v2/permissions/`<codename>`/
- /api/v2/users/
- /api/v2/users/`<username>`/

### PR DESCRIPTION
**Pull request by arsea**
## Description

Modified all URLs listed in #2328 so they used the correct URL. 
The "bug" we discussed in Flowdock ("when I edit a form and I try to open the preview without clicking anythere else before. The preview opens correctly. But if I click on one of the other tabs before opening the preview. the preview fails") should also now work correctly since it uses the correct URL.

## Related issues

Fixes #2328

### CODE REVIEW

File: jsapp/js/dataInterface.es6 [hunk 1]
```diff
@@ -197,7 +197,7 @@ var dataInterface;
 
     getPermissionsConfig() {
       return $ajax({
-        url: `${ROOT_URL}/api/v2/permissions/`,
+        url: `${ROOT_URL}/api/v2/${assetUid}/permissions/`,
```
Comment (magicznyleszek) [inline jsapp/js/dataInterface.es6]: I'm pretty sure this shouldn't have `${assetUid}/` part :)

File: jsapp/js/dataInterface.es6 [hunk 2]
```diff
@@ -258,10 +258,10 @@ var dataInterface;
     assignPerm (creds) {
       // Do we already have these URLs stored somewhere?
       var objectUrl = creds.objectUrl || `${ROOT_URL}/${creds.kind}s/${creds.uid}/`;
-      var userUrl = `${ROOT_URL}/users/${creds.username}/`;
+      var userUrl = `${ROOT_URL}/api/v2/users/${creds.username}/`;
       var codename = creds.role.includes('_submissions') ? creds.role : `${creds.role}_${creds.kind}`;
       return $ajax({
-        url: `${ROOT_URL}/permissions/`,
+        url: `${ROOT_URL}/api/v2/${assetUid}/permissions/`,
```
Comment (magicznyleszek) [inline jsapp/js/dataInterface.es6]: I'm guessing that this also shouldn't have `${assetUid}/` part?

File: jsapp/js/dataInterface.es6 (hunk 1 above)
Comment (magicznyleszek) [reply jsapp/js/dataInterface.es6]: I removed the wrong param from the url :+1:

File: jsapp/js/dataInterface.es6 (hunk 2 above)
Comment (magicznyleszek) [reply jsapp/js/dataInterface.es6]: @noliveleger @arsea I did all changes related to this, @arsea would you like to review the changes?

Comment (jnm) [reply jsapp/js/dataInterface.es6]: I think @arsea might be too busy to look at this; I'll review it now

File: jsapp/js/components/permissions/publicShareSettings.es6 [hunk 3]
```diff
@@ -18,20 +22,27 @@ class PublicShareSettings extends React.Component {
...
-      });
+      if (this.props.kind === 'collection') {
+        actionFn = actions.permissions.assignCollectionPermission;
+      } else {
+        actionFn = actions.permissions.assignAssetPermission;
+      }
+      actionFn(
+        this.props.uid, {
+          user: buildUserUrl(ANON_USERNAME),
+          permission: permConfig.getPermissionByCodename(permRole).url
```
Comment (jnm) [inline jsapp/js/components/permissions/publicShareSettings.es6]: let's change the `permRole` parameter to `permCodename` if that's what it is

**Pull request merged**